src/adapter/src/coord/timeline.rs

```diff
@@ -34,6 +37,29 @@ use crate::coord::{timeline, Coordinator};
 use crate::util::ResultExt;
 use crate::AdapterError;
 
+/// TODO(jkosh44)
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub enum TimelineContext {
+    TimelineIndependent,
+    TimelineDependent,
```

Comment:
These are both "Constant" queries in that both don't depend on sources. One option would be to collapse both of these variants into a single variant called something like `Constant`/`DoesntUseSources`. Then that variant contians another `TimelineDependent`/`TimelineIndependent` enum.

Reply:
I think these should be in a single enum. I'd avoid the word `Constant` because query results can still change based on  the PlanContext (wall time). Maybe:

```
enum TimelineContext {
    TimelineDependent(Timeline),
    TimestampDependent,
    TimestampIndependent,
}
```

Does this make logical sense? They are in order of decreasing dependency. Anything that cares about a timeline MUST also care about a timestamp. But things can not care about a timeline but still care about a timestamp. And there's things that care about neither. That is: timestamp dependency implies timeline dependency.

This would mean that `select mz_now()` would either have to choose a timeline or error. Hmm maybe this is out of scope of this PR then.

Reply:
Wait wait I'm wrong and inconsistent there. I said two competing things in adjacent sentences. Maybe my proposal is: we can never care about a timestamp without a timeline. So maybe this just two items?

Reply:
```
enum TimelineContext {
    TimelineTimestamp(Timeline, Timestamp),
    NoTimestamp,
}
````